File: frontend/types/agent.ts
export interface AgentMeta {
  name: string
  agent_type: "stock" | "portfolio" | "market" | "sector" | string
  description: string
  schedule_cron: string | null
}

export interface AgentListResponse {
  agents: AgentMeta[]
}

export interface AgentReport {
  id: number
  agent_name: string
  agent_type: string
  status: "pending" | "running" | "completed" | "failed"
  target_symbol: string | null
  data: Record<string, unknown> | null
  error_message: string | null
  tokens_used: number | null
  run_duration_ms: number | null
  created_at: string
  updated_at: string | null
}

export interface TriggerResponse {
  status: string
  agent_name: string
  target_symbol: string | null
  message: string
}

export interface SectorSlice {
  sector: string
  total_value: number
  weight_pct: number
  symbols: string[]
  num_holdings: number
}

export interface SectorBreakdown {
  sectors: SectorSlice[]
  total_value: number
}

File: backend/app/schemas/agent.py
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel


class AgentMeta(BaseModel):
    name: str
    agent_type: str
    description: str
    schedule_cron: Optional[str] = None


class AgentListResponse(BaseModel):
    agents: list[AgentMeta]


class AgentReportResponse(BaseModel):
    id: int
    agent_name: str
    agent_type: str
    status: str
    target_symbol: Optional[str] = None
    data: Optional[dict[str, Any]] = None
    error_message: Optional[str] = None
    tokens_used: Optional[int] = None
    run_duration_ms: Optional[int] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class TriggerResponse(BaseModel):
    status: str
    agent_name: str
    target_symbol: Optional[str] = None
    message: str


class SectorSlice(BaseModel):
    sector: str
    total_value: float
    weight_pct: float
    symbols: list[str]
    num_holdings: int


class SectorBreakdownResponse(BaseModel):
    sectors: list[SectorSlice]
    total_value: float
